app/src/main/java/com/example/frogweather/MainActivity.kt

```diff
@@ -80,6 +85,14 @@ class MainActivity : AppCompatActivity() {
         alertDialog.create().show()
     }
 
+    private fun shouldUpdateLocation(): Boolean {
```

Comment:
I don't understand how this method helps you, because the shouldUpdate will always return for the first time false, and also you still listening for the observer, but you are not doing anything with it.